app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java

```diff
@@ -193,4 +211,40 @@ private void getDeviceLocation() {
             }});
         }
     }
+
+    private void setPinniesDrawable(){
+        TypedValue colorValue = new TypedValue();
+        getContext().getTheme().resolveAttribute(com.google.android.material.R.attr.colorOnSecondary, colorValue, true);
+        Drawable unwrapped_pinnies_logo = AppCompatResources.getDrawable(getContext(), R.drawable.ic_pinnies_logo);
+        pinnies_logo = DrawableCompat.wrap(unwrapped_pinnies_logo);
+        DrawableCompat.setTint(pinnies_logo, colorValue.data);
+    }
```

Comment:
Do we have to have this function to set the pinnies drawable? Shouldn't we just be able to specify it in the layout XML?

Reply:
I included this because using the `setCompoundDrawablesWithIntrinsicBounds` function allows the button to resize dynamically once the number of pinnies loads. This is less useful for the constraint layout because we can just create an image view that contains the icon and have that hidden until the Text View is ready so you're probably right